src/render_cmd.rs

```diff
@@ -220,22 +218,17 @@ impl RenderCmd {
     }
 
     /// Utility to turn an option for a vector of pathbufs into a vector of strings
-    /// todo is it really that hard to go from a Vec<PathBuf> to a Vec<String>?
     fn get_values_as_strings(&self, input: &Option<Vec<PathBuf>>) -> Result<Vec<String>> {
```

Comment:
That looks a lot better.

And it got me thinking on how I could express that using functions:

```rust
    fn get_values_as_strings(&self, input: &Option<Vec<PathBuf>>) -> Result<Vec<String>> {
        let mut buffer: Vec<String> = vec![];

        if let Some(items) = input {
            items
                .iter()
                .filter_map(|i| i.to_str())
                .for_each(|s| buffer.push(s.to_string()));
        }
        Ok(buffer)
    }
```

Would the `for` loop be preferred here as it allows for throwing the error? I am just wondering what is more idiomatic.

Reply:
The `for_each` documentation states that a `for` loop is more idiomatic (https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.for_each).
Moreover, the `filter_map` discards the entries that fails to be converted to `str`, while the original code would return an `Err`.

Reply:
Cool, that answers all my question :+1: thanks.

Also the `for` loop is easier to read IMO.